zookeeper-server/src/main/java/org/apache/zookeeper/server/util/DigestCalculator.java

```diff
@@ -37,19 +37,28 @@
 
     // The hardcoded digest version, should bump up this version whenever
     // we changed the digest method or fields.
-    //
-    // Defined it as Integer to make it able to be changed in test via reflection
-    public static final Integer DIGEST_VERSION = 2;
+    private static final int DIGEST_VERSION = 2;
 
+    public static final DigestCalculator DIGEST_CALCULATOR;
     public static final String ZOOKEEPER_DIGEST_ENABLED = "zookeeper.digest.enabled";
-    private static boolean digestEnabled;
 
     static {
-        digestEnabled = Boolean.parseBoolean(
+        boolean digestEnabled = Boolean.parseBoolean(
                 System.getProperty(ZOOKEEPER_DIGEST_ENABLED, "true"));
         LOG.info("{} = {}", ZOOKEEPER_DIGEST_ENABLED, digestEnabled);
+        DIGEST_CALCULATOR = new DigestCalculator(digestEnabled, DIGEST_VERSION);
     }
 
+
```

Comment:
This is not final, why don't you add a setDigestEnabled method?
Make it 'volatile'

We won't lose much in performances